src/core/session.py

```diff
@@ -0,0 +1,32 @@
+# coding: utf8
+
+__contact__ = "info@hytech-imaging.fr"
+__copyright__ = "Copyright (c) 2021 Hytech Imaging"
+
+from qgis.core import QgsProject
+from .database import SammoDataBase
+
+
+class SammoSession:
+    def __init__(self):
+        self.db = SammoDataBase()
+        self.isDbOpened = False
+        self.directoryPath = None
+
+    def isDataBaseAvailable(self, directory):
+        return self.db.isDataBaseAvailableInThisDirectory(directory)
+
+    def createEmptyDataBase(self, directory):
+        self.db.createEmptyDataBase(directory)
+
+    def setDirectoryPath(self, directory):
```

Comment:
no need for a setter. the `directoryPath` member is public.

Reply:
you are right but I prefer not to expose this kind of important variable. Furthermore, we may have to do other thinks (than just change the variable) in the future when the gui/session.py call setDirectorPath.

Do you confirm your want to remove this method ?

Reply:
> you are right but I prefer not to expose this kind of important variable.

I understand your concern but the Python philosophy is slightly different than the C++ one (where a setter would be preferred in this particular case). The pythonic philosophy is expressed by the saying: "We are consenting adults" :).

Reply:
> Do you confirm your want to remove this method ?

Yep

Reply:
Yes you confirm :)

Reply:
Done commit #8 (33fc7e1847da170fdcc5362c7978609aace3250c)